### ISSUE TITLE
Support for compileOnly configuration (gradle-2.12) for querydsl -  lombok and similar

### ISSUE THREAD

**Issue by ieugen** [ewerk/gradle-plugins#39]
Hi, 

I'm trying to compile querydsl classes with JPAAnnotationProcessor. I'm using lombok in my classes (not generated ones).

Up unntil now I'm using the provided scope (from provided-base plugin) to have lombok as compile only but I wish to drop that plugin so  I have a cleaner build and less dependencies.

````
plugins {
    id 'nebula.optional-base' version '3.0.3'
    id 'nebula.provided-base' version '3.0.3'
    id "com.ewerk.gradle.plugins.querydsl" version "1.0.5"
    id 'java'
}
sourceCompatibility = 1.8
targetCompatibility = 1.8

dependencies {
    provided "org.projectlombok:lombok:${lombokVersion}"
    compile "com.mysema.querydsl:querydsl-jpa:${queryDslVersion}"
    compile "org.reflections:reflections:${reflectionsVersion}"
}

querydsl {
    library = "com.mysema.querydsl:querydsl-apt:${queryDslVersion}"
    jpa = true
    querydslSourcesDir = "src/querydsl/java"
}
````

Switching to `compileOnly` instead of provided throws compilation error since lombok is not on querydslCompile classpath.  

I've tried extending querydslCompile from compileOnly but it dod not work. 

Any help is appreciated.  

````
symbol: class ToString
Note: Running JPAAnnotationProcessor
100 errors
:client-api:compileQuerydsl FAILED
* What went wrong:
Execution failed for task ':client-api:compileQuerydsl'.
> Compilation failed; see the compiler error output for details.
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':client-api:compileQuerydsl'.
````

**Comment by holgerstolzenberg** [comment]
Could you please try to extends from `compile`? At the moment the plugin is composed to look at the `compile` classpath only.

The `provided` scope coming from Maven is a thing not supported by Gradle out of the box and is not what you are looking for I think.

The `compileOnly` classpath is new to me. Could you please provide a documentation link about that?

**Comment by ieugen** [comment]
Hi, 

I managed to fix it after reading your comments.  I had to extend from compileClasspath (read the docs to see how configurations are extended [1]). Since I use lombok in tests I added it to the testCompileOnly configuration. 

Thank you,

````
configurations {
    querydsl.extendsFrom compileClasspath
}
dependencies {
    compileOnly "org.projectlombok:lombok:${lombokVersion}"
    testCompileOnly "org.projectlombok:lombok:${lombokVersion}"
}
````
CompileOnly was added to java plugin in gradle 2.12 [1]. It is similar to `provided` . I have replaced it in another project and works like a charm, except with querydsl plugin :( . Lombok is a compileOnly dependency since it does not exist at runtime. 

I think the problem is that the `compileOnly` extends the `compile` configuration  which you extend also. As the docs say, compileClaspath is better IMO. 

[1] https://docs.gradle.org/current/userguide/java_plugin.html
